### ISSUE TITLE
Guest Checkout Issues With Subsequent Purchases

### ISSUE THREAD

**Issue by pevey** [vendure-ecommerce/vendure#3055]
After the first guest purchase, a customer is created, and the customer is considered "logged in" for the remainder of the session but not authorized.  

To be clear, this is not about a customer that registers.  This is all strictly in guest mode.  The checkout process itself creates a customer for guest purchasers.

This creates 2 issues if the cookie is not manually cleared:

- The addPaymentToOrder resolver fails on subsequent attempts because it expects ctx.AuthorizedAsOwner to be true.  In this new state, ctx.AuthorizedAsOwner is false, but ctx.isAuthorized is now strangely true (where before in first guest checkout before the customer was created it was false)

- The setCustomerForOrder resolver also fails with ALREADY_LOGGED_IN_ERROR.  However, despite being pseudo logged in, the transitionOrderToState mutation fails with error "Cannot transition Order to the "ArrangingPayment" state without Customer details."

**Comment by pevey** [comment]
Adding: I think the best way to handle this for now is to make sure when you show your "order success" page, you also use the mutation to log the user out.  I've been testing that, and so far no issues.

**Comment by pevey** [comment]
I just want to add that the workaround above is only for sites that allow guest checkout exclusively.  

For sites that always require logging in (no guest checkout), this issue is never seen.

However, for sites that want to do both guest checkout and allow users to log in, I'm not sure if there is a workaround for this bug.

The first checkout will always work fine.  Every subsequent checkout will fail for a guest customer who has not created an account and authenticated unless the logout mutation is called.  Calling the logout mutation will of course cause the cart to reset.

**Comment by pevey** [comment]
I'm closing this, and I also feel pretty dumb.  I was digging down to try to find the source of why the session got changed after the first purchase when the user was a guest.  I could not find why ctx.AuthorizedAsOwner got changed to false after the first purchase. 

Here is what was actually happening:
After making a purchase, I logged into the admin to see the order go through.  When testing on localhost, even if admin and frontend are on different ports, if the cookie is the same name, it gets overridden at the domain name level, which is localhost for both.  So... that was the issue.
